### PR TITLE
Add "ABGR" format to pygame.image.fromstring

### PR DESCRIPTION
**Pull request by novialriptide**
Ported from: https://github.com/pygame/pygame/pull/3674
Fixes: https://github.com/pygame/pygame/issues/3629

Looking at the SDL_PIXELFORMAT page (https://wiki.libsdl.org/SDL2/SDL_PixelFormatEnum), they have
```
SDL_PIXELFORMAT_RGBA32
SDL_PIXELFORMAT_ARGB32
SDL_PIXELFORMAT_BGRA32
SDL_PIXELFORMAT_ABGR32
```
Similarly, in pygame.image.fromstring, we support
```
RGBA
ARGB
BGRA
```
But with no `ABGR`

Somebody might need that support someday, and it seems like an obvious addition to what we support already.

### CODE REVIEW

**Comment by MyreMylar** [comment]
Debian fail is:

```
====================================================================
  ERROR: test_frombuffer_ABGR (pygame.tests.image_test.ImageModuleTest)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/usr/local/lib/python3.9/dist-packages/pygame/tests/image_test.py", line 1176, in test_frombuffer_ABGR
      argb_surf = pygame.image.frombuffer(abgr_buffer, (4, 4), "ABGR")
  pygame.error: Unknown pixel format
  
  ----------------------------------------------------------------------
  Ran 39 tests in 8.986s
  
  FAILED (errors=1, skipped=1)
  ----------------------------------------------------------------------
```

You might need to do something like this:

```
        try:
            # test if hardware supports ABGR surfaces
            pygame.Surface((64, 64), flags=pygame.SRCALPHA, masks=(0, 0xFF00, 0xFF0000, 0xFF000000))
        except pygame.error:
            pass
        else:
            argb_surf = pygame.image.frombuffer(abgr_buffer, (4, 4), "ABGR")
            self.assertEqual(argb_surf.get_at((0, 0)), pygame.Color(20, 10, 255, 200))
            self.assertEqual(argb_surf.get_at((1, 1)), pygame.Color(255, 255, 255, 127))
            self.assertEqual(argb_surf.get_at((2, 2)), pygame.Color(0, 0, 0, 79))
            self.assertEqual(argb_surf.get_at((3, 3)), pygame.Color(20, 200, 50, 255))
```

If I have those masks correct.

**Comment by novialriptide** [comment]
> You might need to do something like this:

The Debian tests still fail when I insert that code into the tests. Also, I feel like that's a hacky sort of solution. There should be a better way.

File: src_c/image.c [hunk 1]
```diff
@@ -1147,6 +1147,28 @@ image_frombytes(PyObject *self, PyObject *arg)
...
+        for (looph = 0; looph < h; ++looph) {
+            Uint32 *pix = (Uint32 *)DATAROW(surf->pixels, looph, surf->pitch,
+                                            h, flipped);
+            memcpy(pix, data, w * sizeof(Uint32));
+            data += w * sizeof(Uint32);
+        }
+        SDL_UnlockSurface(surf);
+    }
+    else if (!strcmp(format, "ABGR")) {
+        if (len != (Py_ssize_t)w * h * 4)
```
Comment (avaxar) [inline src_c/image.c:1163]: You haven't implemented the changes with the merging of #1863, implementing a `pitch` argument for every format.

**Review by avaxar — CHANGES_REQUESTED** [review]

File: src_c/image.c (hunk 1 above)
Comment (novialriptide) [reply src_c/image.c:1163]: Unfortunate

**Pull request closed without merging**